.github/workflows/cmake.yml

```diff
@@ -0,0 +1,62 @@
+name: CMake
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install boost
+        uses: MarkusJx/install-boost@v2.4.1
+        id: install-boost
+        with:
+          boost_version: 1.73.0
+          
+      - name: Setup NuGet
+        if: ${{ matrix.os=='windows-latest' }}
+        uses: NuGet/setup-nuget@v1.1.1
+        with:
+          nuget-version: '5.x'
+        
+      - name: Install NetworkDirect
+        if: ${{ matrix.os=='windows-latest' }}
+        run: mkdir -p objects/nuget && cd objects/nuget && nuget install NetworkDirect -Version 2.0.1
+      
+      - name: Configure rdma-core
+        if: ${{ matrix.os=='ubuntu-latest' }}
+        run: sudo apt install -y rdma-core librdmacm-dev libibverbs-dev && mkdir -p objects/rdma_core/includes/rdma  && cp -Rf /usr/include/rdma objects/rdma_core/includes/rdma && mkdir -p objects/rdma_core/includes/infiniband && cp -Rf /usr/include/rdma objects/rdma_core/includes/infiniband && cp -Rf /lib/modules/*/build/include/uapi/rdma/* objects/rdma_core/includes/rdma && mkdir -p objects/rdma_core/lib/ && cp -Rf /usr/lib/*/libibverbs.so objects/rdma_core/lib && cp -Rf /usr/lib/*/librdmacm.so objects/rdma_core/lib
```

Comment:
Consider using multiline scripts for readability.
![image](https://user-images.githubusercontent.com/3143819/204938120-b2c25b38-a8e6-4ab9-9a18-c24df4e027da.png)
Source: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun

Since commands won't fail the build then, we can also use "fail-fast" behavior:

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference